command/host_create.go

```diff
@@ -61,8 +66,70 @@ func (c *createHost) Execute(ctx context.Context, comm client.Communicator,
 			return err
 		}
 	}
+	startTime := time.Now()
+	ids, err := comm.CreateHost(ctx, taskData, *c.CreateHost)
+	if err != nil || c.CreateHost.CloudProvider != apimodels.ProviderDocker || c.CreateHost.Background == false {
+		return err
+	}
+	if len(ids) == 0 {
+		return errors.New("Programmer error: no intent host ID received")
+	}
+	if c.CreateHost.StderrFile == "" {
+		c.CreateHost.StderrFile = fmt.Sprintf("container-err-%s.log", ids[0])
+	}
+	if c.CreateHost.StdoutFile == "" {
+		c.CreateHost.StdoutFile = fmt.Sprintf("container-out-%s.log", ids[0])
+	}
+
+	errChan := make(chan error)
+
+	timer := time.NewTimer(time.Duration(c.CreateHost.PollFrequency) * time.Second)
+	defer timer.Stop()
+
+	timeoutTimer := time.NewTimer(time.Duration(c.CreateHost.BackgroundTimeoutSecs) * time.Second)
+	defer timeoutTimer.Stop()
+
+	// get logs in batches until container exits or we timeout
+waitForLogs:
+	for {
+		select {
+		case <-ctx.Done():
+			errChan <- errors.New("context finished waiting for host to exit")
+			break waitForLogs
+		case <-timer.C:
+			curTime := time.Now()
+			logInfo, err := comm.GetDockerLogs(ctx, ids[0], startTime, curTime)
+			startTime = curTime
+			if err != nil || logInfo == nil {
+				errChan <- err
```

Comment:
If `logInfo == nil` but `err == nil`, should this send an error on `errChan` (and not nil)?